### PR TITLE
tsx chat stuff and allow pasting images

### LINKED ISSUE #2989: Should be able to "paste" images into chat.

### PR DESCRIPTION
**Pull request by tscholl2**
This translates `smc_chat` to tsx and attempts to fix #2989.

### CODE REVIEW

**Comment by tscholl2** [comment]
Is this what you had in mind for splitting the commits?

File: src/smc-webapp/smc_chat.tsx [hunk 1]
```diff
@@ -740,296 +835,550 @@ const ChatRoom = rclass(function({ name }) {
...
-      name={name}
-      project_id={project_id}
-      path={path}
-      file_use_id={file_use_id}
-    />
-  );
-};
-
-// Render the given chatroom, and return the name of the redux actions/store
-exports.render = function(project_id, path, dom_node, redux) {
```
Comment (johnjeng) [inline src/smc-webapp/smc_chat.tsx]: Is this no longer used anywhere in our code base?

Comment (williamstein) [reply src/smc-webapp/smc_chat.tsx]: Those were only used before **you** did the top-down react rewrite....

Comment (johnjeng) [reply src/smc-webapp/smc_chat.tsx]: I think some non-react editors opened side chat with this. However I remember now that you changed how side chat opened.

**Pull request merged**